internal/internal_worker.go

```diff
@@ -580,6 +584,9 @@ func (r *registry) RegisterWorkflowWithOptions(
 		}
 	}
 	r.workflowFuncMap[registerName] = wf
+	if options.VersioningBehavior != VersioningBehaviorUnspecified {
+		r.workflowVersioningBehaviorMap[registerName] = options.VersioningBehavior
+	}
```

Comment:
```suggestion
	r.workflowVersioningBehaviorMap[registerName] = options.VersioningBehavior
```

Could just put it in the map always and remove the `, ok` approach from the getter and just rely on unspecified to be equivalent to unset. Not a big deal, but I think it makes for cleaner code.

Reply:
In general I don't like adding unnecessary entries to maps, but here there is no performance/space issue and it is cleaner...
Done